### PR TITLE
{aks} Fix bug related to updating DCR when non monitoring add enabled through az aks enable-addons

### PR DESCRIPTION
**Pull request by ganga1980**
**Related command**

**Description**

Fix will avoid updating existing data collection rules if non monitoring addon trying to enable through
az aks enable-addons

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -3254,3 +3253,23 @@ def _aks_approuting_update(
         return None
 
     return aks_update_decorator.update_mc(mc)
+
+
+def is_monitoring_addon_enabled(addons, instance):
+    is_monitoring_addon_enabled = False
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:3259]: src/azure-cli/azure/cli/command_modules/acs/custom.py:3259:4: W0621: Redefining name 'is_monitoring_addon_enabled' from outer scope (line 3258) (redefined-outer-name)

Comment (ganga1980) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: > src/azure-cli/azure/cli/command_modules/acs/custom.py:3259:4: W0621: Redefining name 'is_monitoring_addon_enabled' from outer scope (line 3258) (redefined-outer-name)

Fixed.

**Review by FumingZhang — APPROVED** [review]

**Comment by yanzhudd** [comment]
please add some test cases for the bug to verify changes.

**Comment by ganga1980** [comment]
> please add some test cases for the bug to verify changes.

added new test to cover this.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**